### PR TITLE
METRON-1026: threatintel_taxii_load.sh throws exception

### PR DESCRIPTION
**Pull request by cestella**
## Contributor Comments
The delegation wrapping done to enable Stellar transformations on data as it is imported broke the taxii loader.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: metron-platform/metron-data-management/src/main/java/org/apache/metron/dataloads/nonbulk/taxii/TaxiiLoader.java [hunk 1]
```diff
@@ -165,6 +167,19 @@ public static Options getOptions() {
...
+    }
+    else if(e instanceof ExtractorDecorator) {
+      ExtractorDecorator decorator = (ExtractorDecorator)e;
+      if(decorator.getUnderlyingExtractor() != null && decorator.getUnderlyingExtractor() instanceof StixExtractor) {
+        return true;
+      }
+    }
+    return false;
+  }
+
```
Comment (simonellistonball) [inline metron-platform/metron-data-management/src/main/java/org/apache/metron/dataloads/nonbulk/taxii/TaxiiLoader.java]: why not just?
`e instanceof StixExtractor || (ExtractorDecorator)e.getUnderlyingExtractor() instanceof StixExtractor`
this is java after all and has shortcutting... even stellar does that now too :)

Comment (cestella) [inline metron-platform/metron-data-management/src/main/java/org/apache/metron/dataloads/nonbulk/taxii/TaxiiLoader.java]: good catch :)

**Comment by cestella** [comment]
Note for testing this PR, I found the easiest way to install opentaxii is via the opentaxii role.  Unfortunately `run_ansible_role.sh opentaxii` did not work for me, so I resorted to modifying `metron-deployment/playbooks/metron_install.yml` and adding 
```
   - { role: opentaxii,              tags: ['always'] }
```
under the monit hosts section.  This is totally a hack and I would love to hear a better way to run the opentaxii role on a working full-dev.

**Comment by cestella** [comment]
Presumptions:
* Fulldev has opentaxii installed with the `guest.phishtank_com` collection configured

Test:

* Ensure that opentaxii is running by running `service opentaxii status`
* Ensure that the collection is sync'd by running `service opentaxii sync guest.phishtank_com 2017-08-01`
* Open `~/taxii.json` and input the following:
```
{
   "endpoint" : "http://localhost:9000/services/discovery"
  ,"type" : "DISCOVER"
  ,"collection" : "guest.phishtank_com"
  ,"table" : "threatintel"
  ,"columnFamily" : "t"
  ,"allowedIndicatorTypes" : [ ]
}
```
* Create a file `~/extractor.json` with the following:
```
{
  "extractor" : "STIX",
  "config" : {}
}
```
* Run the taxii loader via `/usr/metron/0.4.1/bin/threatintel_taxii_load.sh -e ./extractor.json -c ./taxii.json`
* Ensure it functions and data flows as usual (or not, depending on what data is in hailataxii).  Prior to this PR, you'd have seen an exception.

One thing to note, most of the recent hailataxii data are URIs which isn't something that we support.  I patched this branch with support for URIs and verified data was flowing into HBase.  That URI support is done via #689

**Comment by james-sirota** [comment]
+1

**Pull request closed without merging**